stdlib/public/core/Array.swift

```diff
@@ -1202,6 +1202,47 @@ extension Array: RangeReplaceableCollection {
     _appendElementAssumeUniqueAndCapacity(oldCount, newElement: newElement)
     _endMutation()
   }
+  
+  /// Adds the elements of a collection to the end of the array.
+  ///
+  /// Use this method to append the elements of a collection to the end of this
+  /// array. This example appends the elements of a `Range<Int>` instance
+  /// to an array of integers.
+  ///
+  ///     var numbers = [1, 2, 3, 4, 5]
+  ///     numbers.append(contentsOf: 10...15)
+  ///     print(numbers)
+  ///     // Prints "[1, 2, 3, 4, 5, 10, 11, 12, 13, 14, 15]"
+  ///
+  /// - Parameter newElements: The elements to append to the array.
+  ///
+  /// - Complexity: O(*m*) on average, where *m* is the length of
+  ///   `newElements`, over many calls to `append(contentsOf:)` on the same
+  ///   array.
+  @inlinable @_alwaysEmitIntoClient
+  @_semantics("array.append_contentsOf")
+  @_effects(notEscaping self.value**)
+  public mutating func append(contentsOf newElements: __owned some Collection<Element>) {
```

Comment:
Do we _really_ need a new public entry point for this? I.e., does this provide meaningful benefits that aren't already provided by the existing entry point?

Reply:
My thinking is that it can help mitigate the code size increase, because in the (extremely common) case where it's being called with a Collection, we can avoid inlining the underestimate-handling code.

It's possible DCE was already taking care of this satisfactorily though.